src/rs232.rs

```diff
@@ -105,10 +105,26 @@ impl RS232 {
 
         for _ in 1..3 {
             if let Ok(_) = self.read_line(& mut buffer) {
+                // Galieo
+                if buffer.starts_with("$GA") {
+                    return true;
+                }
+                // BeiDou
+                if buffer.starts_with("$GB") {
+                    return true;
+                }
+                // GLONASS
+                if buffer.starts_with("$GL") {
+                    return true;
+                }
+                // GNSS mode (combined/mixed sources)
+                if buffer.starts_with("$GN") {
+                    return true;
+                }
+                // GPS, SBAS, QZSS
```

Comment:
Cool, good change but I think we should just have an array with all the codes and then loop over them.

Reply:
Why?

Reply:
To separate out the logic from the data and avoid unneeded repetition.

Reply:
oh and we should put the array as a module-wide constant:

const TALKER_IDS: &'static [&'static str] = &[
    "$GP", // GPS, SBAS, QZSS
    "$GN", // GNSS mode (combined/mixed sources)
...
};

Reply:
Actually, testing the Talker IDs are unnecessary. I focused on that last night, but it’s actually not necessary. I’ll change the test to check that it resembles an NMEA message structure using the following criteria:

* buffer length is at least 10 characters
* starts with "$G"
* the sixth character is ","

I’ll also increase the number of lines tested to 5 as some devices send proprietary protocols and NMEA intermixed. Just had my device randomly fail testing because the 3 first lines were of a proprietary protocol before the NMEA protocol appeared. My logs shows no more than 3 lines of the proprietary protocol at a time, not sure if other devices behave the same.

Reply:
Hmm.. true. My intention with this code was not to check the IDs really but more like ensuring that we are reading the serial device correctly. The IDs were the easiest to check for but I'm alll for improving it and making it more generic.

Reply:
I haven't seen any proprietary data from any device yet but I've only tested against two (old) devices.